services/src/main/scala/cromwell/services/healthmonitor/HealthMonitorServiceActor.scala

```diff
@@ -32,49 +34,71 @@ trait HealthMonitorServiceActor extends Actor with LazyLogging with Timers {
     subsystems.map((_, CachedSubsystemStatus(UnknownStatus, now))).toMap
   }
 
+  subsystems foreach { s => self ! Check(s, failureRetryCount) }
+
+  private def check(subsystem: MonitoredSubsystem, after: FiniteDuration, withRetriesLeft: Int): Unit = {
+    timers.startSingleTimer(UUID.randomUUID(), Check(subsystem, withRetriesLeft), after)
+  }
+
+  private def scheduleFailedRetryCheck(subsystem: MonitoredSubsystem, retriesLeft: Int): Unit = {
+    check(subsystem, after = failureRetryInterval, withRetriesLeft = retriesLeft - 1)
+  }
+
+  private[healthmonitor] def scheduleSweepCheck(subsystem: MonitoredSubsystem): Unit = {
+    check(subsystem, after = sweepInterval, withRetriesLeft = failureRetryCount)
+  }
+
   override def receive: Receive = {
-    case CheckAll =>
-      logger.debug("Checking status on all subsystems")
-      subsystems.foreach(checkSubsystem)
-    case Store(subsystem, status) => store(subsystem, status)
+    case Check(subsystem, retriesLeft) =>
+      checkSubsystem(subsystem, retriesLeft)
+    case Store(subsystem, status) =>
+      store(subsystem, status)
+      scheduleSweepCheck(subsystem)
     case GetCurrentStatus => sender ! getCurrentStatus
     case ShutdownCommand => context.stop(self) // Not necessary but service registry requires it. See #2575
     case Status.Failure(f) => throw f
-    case _ => throw new RuntimeException("wtf")
+    case e => throw new RuntimeException(e.toString)
   }
 
-  private def checkSubsystem(subsystem: MonitoredSubsystem): Unit = {
+  private def checkSubsystem(subsystem: MonitoredSubsystem, retriesLeft: Int): Unit = {
     val result = subsystem.check()
     val errMsg = s"Timed out after ${futureTimeout.toString} waiting for a response from ${subsystem.toString}"
-    result.withTimeout(futureTimeout, errMsg, context.system.scheduler)
-      .recover { case NonFatal(ex) =>
-        failedStatus(ex.getMessage)
-      } map {
-      Store(subsystem, _)
-    } pipeTo self // Piping this instead of calling store() directly so as to not modify actor state inside the Future
-
+    result.withTimeout(futureTimeout, errMsg, context.system.scheduler) onComplete {
+      case Success(status) if status.ok =>
+        self ! Store(subsystem, status)
+      case Success(_) if retriesLeft > 0 =>
+        // This is a success in getting the status but the status is a failure. However there are retries available.
+        scheduleFailedRetryCheck(subsystem, retriesLeft)
+      case Success(status) =>
+        // This is also a success in getting the status but the status is a failure and there are no more retries available.
+        // Womp womp.
+        self ! Store(subsystem, status)
+      case Failure(NonFatal(_)) if retriesLeft > 0 =>
+        scheduleFailedRetryCheck(subsystem, retriesLeft)
+      case Failure(NonFatal(e)) =>
+        self ! Store(subsystem, SubsystemStatus(ok = false, messages = Option(List(e.getMessage))))
+      case f =>
+        self ! f
```

Comment:
ToL: maybe wrap that into a `Status.Failure(f)` or explicitly handle the case of a `scala.Failure` in the `receive` method ?

Reply:
At this point in the match `f` could only be a `scala.Failure` already and is explicitly "handled" in the `receive` method:

```
    case Status.Failure(f) => throw f
```
🙄

Reply:
Ah right 😄